src/main/java/org/opentripplanner/api/model/Place.java

```diff
@@ -104,6 +105,26 @@
      */
     public String bikeShareId;
 
+    /**
+     * Type of board or alight (to distinguish flexible transit modes).
+     */
+    public BoardAlightType boardAlightType;
+
+    /**
+     * Board or alight area for flag stops
+     */
+    public EncodedPolylineBean flagStopArea;
+
+    /**
+     * The longitude of the route geometry endpoint for a deviated-route place.
+     */
+    public Double deviatedRouteLon = null;
+
+    /**
+     * The latitude of the route geometry endpoint for a deviated-route place.
+     */
+    public Double deviatedRouteLat = null;
+
```

Comment:
I agree that this is kind of awkward from an API design perspective. The issue is that the deviated segment of a transit leg is not itself a separate leg. In the most complicated case, a vehicle could deviate from its route to pick a passenger up, go back to its route, and then deviate again for the dropoff.

This is handled in the API as follows: it's all within a single Leg in the API response, since from the passenger's perspective, it _is_ a single leg. leg.geometry contains the geometry of the vehicle during the portion its on it's scheduled route. `leg.from` is the boarding location and `leg.to` is the  alighting location, which are both `Places`; `place.deviatedRouteLat/place.deviatedRouteLon` are the locations on the route geometry where the vehicle rejoins the scheduled route. Arguably, this is redundant, since its either the first or last point of `leg.geometry`, but in the UI development, it was useful to have this.

Does this make sense? We could do a couple things:
- leave as is
- set `leg.geometry` to include the deviated portions (the OSM segments that OTP infers the vehicle travels on). I don't recommend this, since the exact route on a deviated segment isn't really known, it's just a guess.
- Remove `deviatedRouteLon/deviatedRouteLat` since they are redundant.
- some other API solution?

Any thoughts?

Reply:
What I ment was that you could use composition. Put these 4 properties into a class(lets call it `PointFlexExtention`) and add `public PointFlexExtention flex`to the Place class. 
Pros:
- For all places that have nothing to do with flex you only have one empty property `flex=null`
- The concept that these parameters relates to each other is clearly visible.
Cons:
- The model get deeper/more nesting

Reply:
Unfortunately the API is not so much designed as it is grown and evolved within a set of constraints. Ideally as we move toward 2.0 though we will become more principled about API design. Since @sdjacobs has stated that this new field in the trip planning results is redundant (in practice it is always the first or last point in the leg shape) I would prefer to remove it.

Reply:
Will do.